### PR TITLE
fluent-bit: add package

### PR DESCRIPTION
**Pull request by rthakur33**
* add package fluent-bit v3.1.0

### CODE REVIEW

**Comment by krant** [comment]
There is no need to specify all CMake options matching the defaults - it makes Makefile larger and harder to maintain.

**Comment by rthakur33** [comment]
> There is no need to specify all CMake options matching the defaults - it makes Makefile larger and harder to maintain.

Thanks for your comment, it is addressed now.

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:7]: This is not correct. We can not download files "vVERSION", because in the dl folder it is going to be mess and we would not be sure to which package does it belong, yeah, you can verify it manually, but you probably won't verify it, if it was not overwritten by the other package with the same version. 

So, it should be as usual as it is done in other Makefiles - packageName-Version.

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:22]: Two spaces as indentation here.

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:26]: Two spaces here as well.

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:45]: We can order it a little bit more.

First folder, then file.

Also, the config file should be INSTALL_CONF and shouldnt it be marked as conffile to prevent to be overwritten by update?

**Review by BKPepe — CHANGES_REQUESTED** [review]
In the pull request and as well in the commit description, I do miss some description. If the package was compile tested, run tested and also why do we need to add this.

1. This goes into admin folder, but you do have category Utilities in the Makefile.
I suggest to look at it more, where it belongs, because it is also missing Section in the define.

**Comment by rthakur33** [comment]
> In the pull request and as well in the commit description, I do miss some description. If the package was compile tested, run tested and also why do we need to add this.
> 
> 1. This goes into admin folder, but you do have category Utilities in the Makefile.
>    I suggest to look at it more, where it belongs, because it is also missing Section in the define.

Thanks for reviewing this, I have addressed all your comments in the update pull request.

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: done

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: done

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: done

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: done

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:11]: Please don't do this. This is even mentioned in the documentation, how you should not do it. I think this says it all. Skipping something is not wanted not even for checking checksum. Not sure, where did you find it.

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:10]: Here you are mixing at least it looks like two things. Checking git sources and downloading tarball. You should decide which one would you want to use.

Comment (rthakur33) [reply admin/fluent-bit/Makefile]: yes, I realise, updated, thanks!

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: sorry, updated

File: admin/fluent-bit/Makefile
Comment (1715173329) [inline admin/fluent-bit/Makefile:23]:
```suggestion
  TITLE:=Fast and Lightweight Logs and Metrics processor
  URL:=https://fluentbit.io/
  DEPENDS:=+libyaml +libopenssl +libcurl +libatomic +musl-fts +flex +bison
```

File: admin/fluent-bit/Makefile
Comment (1715173329) [inline admin/fluent-bit/Makefile:27]:
```suggestion
  Fluent Bit is a super fast, lightweight, and highly scalable logging
  and metrics processor and forwarder.
```

wrap a new line for every ~70 chars

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: thanks for the review, done

File: admin/fluent-bit/Makefile
Comment (rthakur33) [reply admin/fluent-bit/Makefile]: thanks for reviewing, done.

**Review by 1715173329 — APPROVED** [review]

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:4]: Before merging this, can we update this to 3.1.3, please? :)

File: admin/fluent-bit/Makefile
Comment (BKPepe) [inline admin/fluent-bit/Makefile:35]: Do we need this comment?
```suggestion
```

**Review by BKPepe — APPROVED** [review]
Otherwise 👍 Thank you!

**Comment by rthakur33** [comment]
> Otherwise 👍 Thank you!

thanks for the review, comments addressed.

**Comment by BKPepe** [comment]
Two approvals, two reviewers, so many reviews, no other choice than merging this. :-)

**Comment by hnyman** [comment]
Seems to fail in buildbot due to dependencies, as some libraries are found...
(and seems to vary between targets...)

https://downloads.openwrt.org/snapshots/faillogs/aarch64_cortex-a72/packages/fluent-bit/compile.txt

> 
> Package fluent-bit is missing dependencies for the following libraries:
> libz.so.1
> libzstd.so.1

https://downloads.openwrt.org/snapshots/faillogs/mips_24kc/packages/fluent-bit/compile.txt
https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/fluent-bit/compile.txt
> Package fluent-bit is missing dependencies for the following libraries:
> libsasl2.so.3
> libz.so.1
> libzstd.so.1

**Comment by krant** [comment]
Adding this to CMAKE_OPTIONS must fix the issue:
```
	-DWITH_ZLIB=OFF \
	-DWITH_ZSTD=OFF \
	-DWITH_SASL=OFF \
```

**Pull request merged**
